### PR TITLE
Rework readme and example/readme

### LINKED ISSUE #135: Readme/example improvements
Follow up from: #130 

We should improve README.md as shown on GitHub and pub.dev and the [example tab on pub.dev](https://pub.dev/packages/objectbox/example).

Maybe take a clue from the most liked packages on pub.dev
* https://pub.dev/packages/provider/example 
* https://pub.dev/packages/shared_preferences
* https://pub.dev/packages/get

### PR DESCRIPTION
**Pull request by vaind**
Closes #135 

I've basically rewritten both README.md files so I suggest not looking at the changes but rather at final files themselves.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,194 +1,115 @@
...
-var box = Box<Note>(store);
-    
-var note = Note(text: "Hello");
-note.id = box.put(note);
-print("new note got id ${note.id}");
-print("refetched note: ${box.get(note.id)}");
-```
+* 🏁 **High performance** - improving response rates and enabling real-time applications.
+* 🪂 **ACID compliance** - Atomic, Consistent, Isolated, Durable.
+* 🔗 **Relations** - object links / relationships are build-in.
```
Comment (greenrobot-team) [inline README.md:10]:
```suggestion
* 🔗 **Relations** - object links / relationships are built-in.
```

File: README.md [hunk 2]
```diff
@@ -1,194 +1,115 @@
...
+  ```
 
-    // clean up
-    sub2.cancel();
+* macOS: dart later complain that it cannot find the `libobjectbox.dylib`. You probably have to unsign the `dart`
+  binary - see [this dart issue](https://github.com/dart-lang/sdk/issues/38314#issuecomment-534102841) for details.
 
-    store.close();
-```
+* Windows: copy the downloaded `lib/objectbox.dll` to `C:\Windows\System32\` (requires admin privileges).
```
Comment (greenrobot-team) [inline README.md:94]: Doesn't this lead to a versioning nightmare? E.g. all ObjectBox projects on that machine would need to use that version. And how to determine what version `objectbox.dll` is?

Comment (vaind) [reply README.md:94]: Yes. That's the case with objectbox-c until 1.0 release and ABI stability.

File: README.md (hunk 1 above)
Comment (vaind) [reply README.md:10]: of course you're right... copy paste from https://objectbox.io/dev-get-started/

File: example/README.md [hunk 3]
```diff
@@ -1,88 +1,223 @@
...
+Then, we define a single "Entity" that should be persisted by ObjectBox. You can have multiple
+entities in the same file, or you can have them spread across multiple files in your package's `lib` directory.
+
+```dart
+import 'package:objectbox/objectbox.dart';
+
+@Entity()
+class Note {
+    // Each "Entity" needs a unique integer ID property.
+    // Add `Id()` annotation if its name isn't "id" (case insensitive).
```
Comment (greenrobot-team) [inline example/README.md:21]:
```suggestion
    // Add `@Id()` annotation if its name isn't "id" (case insensitive).
```
Assuming `@` is required.

File: example/README.md [hunk 4]
```diff
@@ -1,88 +1,223 @@
...
+
+    // An empty default constructor is needed but you can use optional args.
+    Note({this.text});
+
+    // Note: just for logs in the examples below(), not needed by ObjectBox.
+    toString() => 'Note{id: $id, text: $text}';
+}
+```
+
+To generate ObjectBox binding code for your entities, run `(flutter|dart) pub build_runner build`.
```
Comment (greenrobot-team) [inline example/README.md:39]: `(flutter|dart) pub build_runner build` for a novice, isn't this confusing? (Or does this or syntax actually work?)

File: example/README.md [hunk 5]
```diff
@@ -1,88 +1,223 @@
...
+
+Use `getApplicationDocumentsDirectory()` from the `path_provider` package to retrieve this directory.
 
-To configure ObjectBox properly, you can use `getApplicationDocumentsDirectory()` from the `path_provider` package. 
-See [Flutter: read & write files](https://flutter.dev/docs/cookbook/persistence/reading-writing-files) for more info. 
-Have a look how it's done in the Flutter example app:
 ```dart
 import 'package:path_provider/path_provider.dart';
 
+import 'objectbox.g.dart'; // created by `pub run build_runner build`
```
Comment (greenrobot-team) [inline example/README.md:63]: Hm, above the command was `(flutter|dart) pub build_runner build`, am I missing something?

File: example/README.md [hunk 6]
```diff
@@ -1,88 +1,223 @@
...
+directory, but it doesn't have permissions to write there: `failed to create store: 10199 Dir does not exist: objectbox (30)`.
+
+### Dart CLI apps
+
+In standard apps (not on mobile), ObjectBox can write anywhere the current user can. The following minimal example
+omits the argument to `Store(directory: )`, thus using the default - 'objectbox' in the current working directory.
 
 ```dart
-import "package:objectbox/objectbox.dart";
+import 'objectbox.g.dart'; // created by `pub run build_runner build`
```
Comment (greenrobot-team) [inline example/README.md:96]: Hm, above the command was `(flutter|dart) pub build_runner build`, am I missing something?

**Review by greenrobot-team — COMMENTED** [review]
Yay, new READMEs! Found some potential issues.

File: example/README.md (hunk 5 above)
Comment (vaind) [reply example/README.md:63]:
```suggestion
import 'objectbox.g.dart'; // created by `flutter pub run build_runner build`
```

File: example/README.md (hunk 6 above)
Comment (vaind) [reply example/README.md:96]:
```suggestion
import 'objectbox.g.dart'; // created by `dart pub run build_runner build`
```

File: example/README.md (hunk 4 above)
Comment (vaind) [reply example/README.md:39]:
```suggestion
To generate ObjectBox binding code for your entities, run `pub run build_runner build`.
```

**Pull request merged**